homeassistant/components/onkyo/config_flow.py

```diff
@@ -67,6 +69,40 @@ class OnkyoConfigFlow(ConfigFlow, domain=DOMAIN):
     _receiver_info: ReceiverInfo
     _discovered_infos: dict[str, ReceiverInfo]
 
+    async def async_step_ssdp(
+        self, discovery_info: ssdp.SsdpServiceInfo
+    ) -> ConfigFlowResult:
+        """Handle flow initialized by SSDP discovery."""
+        host = urlparse(discovery_info.ssdp_location).hostname
+        _LOGGER.debug("Config flow start ssdp auto discovery: %s", host)
+
+        if not host:
+            _LOGGER.error("SSDP host is None: %s", discovery_info)
+            return self.async_abort(reason="no_host")
+
+        try:
+            info = await async_interview(str(host))
```

Comment:
I'm not sure we want to make a connection each time we get a ssdp discovery.

Reply:
Discussed on slack, 

"set unique id to last chunk of UUID, call abort if already configured, then if not aborted, run network command to load serial number, reset unique id to new serial number, and call abort if configured again"